Application/EdFi.Ods.Features/Profiles/ProfileContentTypeContextMiddleware.cs

```diff
@@ -0,0 +1,111 @@
+        {
+            // Only process profile content types
+            string profileContentType = context.Request.ContentType;
+
+            if (profileContentType?.StartsWith(ProfileContentTypePrefix) ?? false)
+            {
+                ProcessProfileContentType("Content-Type", profileContentType);
+            }
+        }
+
+        await _next(context);
+
+        void ProcessProfileContentType(string headerName, string contentType)
+        {
+            // e.g. application/vnd.ed-fi.student.test-profile.readable+json
+            if (MediaTypeHeaderValue.TryParse(contentType, out var mt))
+            {
+                // Skip known existing segments for "vnd.ed-fi", and retrieve just the resource name, profile name and usage
+                var profileContentTypeFacets = mt.Facets.Skip(2).ToArray();
+
+                if (profileContentTypeFacets.Length != 3)
+                {
+                    throw new BadRequestException($"The format of the profile-based '{headerName}' header was invalid.");
+                }
+
+                // SPIKE NOTE: Probably will want to remove this in favor of the same check elsewhere before final merge. Just need to make sure that a check is performed that the resource in the profile content matches the resource in the request
+                // TODO: Ensure that the resource name of the profile request context matches the name of the resource from the DataManagementResourceContext - paying attention to the order of the middleware 
+                // if (!profileRequestContext.ResourceName.EqualsIgnoreCase(resourceClassFullName.Name))
+                // {
+                //     throw new BadRequestException(
+                //         "The resource specified in the profile-based content type did not match the resource targeted by the request.");
+                // }
+                
+                profileContentTypeContextProvider.Set(
+                    new ProfileContentTypeContext(
+                        profileContentTypeFacets[ProfileNameFacet].Value,
+                        profileContentTypeFacets[ResourceNameFacet].Value,
+                        GetContentTypeUsage(headerName, profileContentTypeFacets[UsageFacet])));
+            }
+        }
+
+        ContentTypeUsage GetContentTypeUsage(string headerName, StringSegment profileContentTypeFacet)
+        {
+            var usageSegment = profileContentTypeFacet;
+
+            if (usageSegment.Equals("readable", StringComparison.OrdinalIgnoreCase))
+            {
+                return ContentTypeUsage.Readable;
+            }
+            else if (usageSegment.Equals("writable", StringComparison.OrdinalIgnoreCase))
+            {
+                return ContentTypeUsage.Writable;
+            }
+   
+            throw new BadRequestException($"The profile usage segment in the profile-based '{headerName}' header was not recognized.");
```

Comment:
This returns HTTP 500 instead of 400. I didn't test the other errors but most likely have the same behaviour.
![image](https://user-images.githubusercontent.com/8497432/197083111-c05cb569-86f1-42d9-9295-27a6085d7978.png)

Reply:
So the work to address this goes a little deeper than we need to for this spike ticket. This middleware is running at a much lower level in the processing than the fallback exception translation (the `ExceptionHandlingFilter`).

I've added [a comment](https://tracker.ed-fi.org/browse/ODS-5610?focusedCommentId=70867&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-70867) to the ticket for updating the Profiles Postman tests to ensure we don't forget to address this when it comes time to work on the ticket to prepare the branch for final merge to `main` with [ODS-5611](https://tracker.ed-fi.org/browse/ODS-5611).

Reply:
I have added Postman test in the "Profiles" test suite covering this scenario, expecting a 400 with the exception message in the response.

Reply:
Covering tests have been moved into the main Profiles Suite now, and are passing (with commits added to ODS-5610 branch, which is based on this one).